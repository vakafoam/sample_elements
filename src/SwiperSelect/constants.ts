// default item width in px
export const ITEM_WIDTH: number = 100;

export interface SettingsType {
  width?: string;
  height?: string;
  backgroundColor?: string;
  fontColor?: string;
  activeFontColor?: string;
  fontFamily?: string;
  fontSize?: string;
  units?: string;
}

export const SETTINGS: SettingsType = {
  width: "80%",
  height: "100px",
  backgroundColor: "#285194",
  fontColor: "white",
  activeFontColor: "white",
  fontFamily: "Montserrat,sans-serif",
  fontSize: "24px",
  units: "â‚¬/Month",
};
