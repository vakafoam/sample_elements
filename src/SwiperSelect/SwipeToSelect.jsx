import React from "react";
import styled from "styled-components";
import { ITEM_WIDTH, SETTINGS } from "./constants";

const SelectWrapper = styled.section`
  // display: flex;
  width: 80%;
  position: relative;
  overflow: hidden;
  margin: auto;
  background: black;
`;

const SelectContent = styled.div`
  display: flex;
  position: relative;
  justify-content: flex-start;
  flex-flow: row nowrap;
  flex-shrink: 1;
  flex-grow: 0;
  background: pink;
  overflow: hidden;
  left: calc(50% - ${ITEM_WIDTH / 2}px); // active item in the middle
`;

const SelectOptions = styled.div`
  display: flex;
  position: relative;
  background: white;
  //   align-items: center;
  cursor: pointer;
  justify-content: center;

  // From info element
  position: relative;
  //   display: block;
  flex-direction: column;
  background: yellow;
  height: 80px;
  flex-shrink: 0;
  width: ${ITEM_WIDTH}px;
`;

const TimelineInfo = styled.div`
  position: relative;
  display: block;
  background: yellow;
  height: 100%;
  flex-shrink: 0;
  width: ${ITEM_WIDTH}px;
`;

const SelectedTag = styled.span`
  position: absolute;
  width: 100%;
  top: 5px;
  display: block;
  justify-self: flex-start;
  text-align: center;
  font-size: 12px;
`;

const SelectOption = styled.p`
  box-sizing: border-box;
  justify-self: center;
  line-height: 1;
  text-align: center;
  position: relative;
  font-size: ${({ isActive }) => (isActive ? "26px" : "24px")};
  font-weight: 400;
  margin: 0;
  color: ${({ isActive }) => (isActive ? "blue" : "black")};
  transition: font-size 0.1s ease-out;
  transition-delay: 0.2s;
`;

class SwipeToSelect extends React.Component {
  state = {
    itemsNum: 0,
    i: 0,
    x0: null,
    tx: 0,
    isDragging: false,
    selected: null,
  };

  componentDidMount() {
    const { options } = this.props;
    this.findSelectedItem(options);
    if (options.length) {
      this.setState({ itemsNum: options.length });
    }
  }

  findSelectedItem = arr => {
    let selected = arr.findIndex(el => {
      if (el.selected) return true;
    });
    // TODO: implement initial value from options
    if (selected === -1) selected = 0;
    console.log("selected", selected);
    this.setState({ ...this.state, i: selected, selected });
  };

  // Lock the Timeline from moving to the next slide while user drags
  lock = e => {
    const x0 = this.unify(e).clientX;
    this.setState({ x0, isDragging: true });
  };

  //   unlock = e => {
  //     const x0 = this.unify(e).clientX;
  //     this.setState({ x0, isDragging: false });
  //   };

  isItemActive = i => {
    const el = document.getElementById(`item_${i}`);
    console.log("Element", el && el.getBoundingClientRect());
  };

  move = (e, idx = null) => {
    const { x0, i, itemsNum, isDragging } = this.state;
    const dx = this.unify(e).clientX - x0;
    if (idx !== null && Math.abs(dx) < 5) {
      // handle only click
      console.log("IDX", idx);
      this.setState(prevState => {
        return { ...prevState, i: idx, isDragging: false };
      });
    } else if (isDragging) {
      const s = Math.round((dx * 1) / ITEM_WIDTH);

      if ((i > 0 || s < 0) && (i < itemsNum - 1 || s > 0)) {
        const newI = i - s < 0 ? 0 : i - s >= itemsNum ? itemsNum - 1 : i - s; // not sliding beyond min-max values
        this.setState({ i: newI, tx: 0, isDragging: false, x0: null });
      }
      this.setState({ tx: 0, x0: null, isDragging: false });
    }
  };

  // Get the right event object depending on if it's Touch or Mouse
  unify = e => {
    return e.changedTouches ? e.changedTouches[0] : e;
  };

  // Manage the UI response while user drags the Timeline
  drag = e => {
    const { x0, isDragging } = this.state;

    if (isDragging) {
      console.log("EVENT", e);
      if (!e.changedTouches) e.preventDefault();
      e.stopPropagation();

      const tx = this.unify(e).clientX - x0;
      this.setState({ tx });
    }
  };

  updateItem = (e, idx) => {
    const { x0, i, itemsNum, isDragging } = this.state;

    e.stopPropagation();
    // TODO: check if dragging, then don't update
    const dx = this.unify(e).clientX - x0;
    console.log("click: ", dx);

    if (idx && Math.abs(dx)) {
      // handle only click
      console.log("IDX", idx);
      this.setState(prevState => {
        return { ...prevState, i: idx, isDragging: false };
      });
    }

    //   this.setState({ i: index  })
  };

  onKeyDown = e => {
    const { i, itemsNum } = this.state;
    if (e.keyCode === 37 || e.keyCode === 40) {
      if (i === 0) return;
      this.setState(prevState => {
        return { ...prevState, i: prevState.i - 1 };
      });
    } else if (e.keyCode === 38 || e.keyCode === 39) {
      if (i === itemsNum - 1) return;
      this.setState(prevState => {
        return { ...prevState, i: prevState.i + 1 };
      });
    }
  };

  render() {
    const { i, itemsNum, tx, isDragging, selected } = this.state;
    const { options } = this.props;

    const transformValue = `translate(${(i * -100) /
      itemsNum}%) translate(${tx}px)`;

    const contentStyles = {
      width: `${itemsNum * ITEM_WIDTH}px`,
      transform: transformValue,
      transition: isDragging ? "none" : "transform 0.5s ease-out",
    };
    const translatePoints = -((i - 2) / itemsNum) * 100;

    return (
      <SelectWrapper>
        <SelectContent
          onMouseDown={this.lock}
          //   onMouseUp={this.move}
          // onMouseUp={this.unlock}
          onMouseMove={this.drag}
          onTouchStart={this.lock}
          onTouchEnd={this.move}
          onTouchMove={this.drag}
          onKeyDown={this.onKeyDown}
          onMouseLeave={this.move}
          style={contentStyles}
          tabIndex="0"
        >
          {options.map(({ value, text }, idx) => (
            <SelectOptions id={`item_${idx}`} key={value}>
              {/* <TimelineInfo> */}
              {selected === i && i === idx && (
                <SelectedTag>Current choice</SelectedTag>
              )}
              <SelectOption
                onClick={e => this.move(e, idx)}
                onDoubleClick={e => {
                  e.preventDefault();
                  this.move(e, idx);
                }}
                isActive={i === idx}
                // isActive={this.isItemActive(idx)}
              >
                {text}
              </SelectOption>

              {/* </TimelineInfo> */}
            </SelectOptions>
          ))}
        </SelectContent>
        {/* <TimelinePointsWidget
          i={i}
          translatePoints={
            translatePoints < 0
              ? translatePoints >= -50
                ? translatePoints
                : -50
              : 0
          }
          timelineYears={options.map(item => item.year)}
          updateItem={this.updateItem}
        /> */}
      </SelectWrapper>
    );
  }
}

export default SwipeToSelect;
