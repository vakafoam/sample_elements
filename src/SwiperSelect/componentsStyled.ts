import styled from "style/styled-components";
import { SettingsType } from "SwiperSelect";
import { ITEM_WIDTH } from "SwiperSelect/constants";

export const SelectedState = styled.div<{ settings: SettingsType }>`
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  width: ${({ settings }) => settings.width};
  height: ${({ settings }) => settings.height};
  background: ${({ settings }) => settings.backgroundColor};
  color: ${({ settings }) => settings.fontColor};
  font-size: ${({ settings }) => settings.fontSize};
  margin: auto;
  padding: 10px;
  border-radius: 5px;
`;

export const SelectedValue = styled.span<{ settings: SettingsType }>`
  display: flex;
  flex-direction: column;
  text-align: left;
  justify-content: center;
  margin-left: ${({ settings }) => settings.fontSize};
`;

export const ValueUnits = styled.span<{ settings: SettingsType }>`
  font-size: calc(${({ settings }) => settings.fontSize} - 8px);
  margin-top: 5px;
`;

export const EnableEdit = styled.button<{ settings: SettingsType }>`
  position: relative;
  display: block;
  justify-self: flex-end;
  align-self: center;
  border: none;
  padding: 20px;
  background-color: transparent;
  color: ${({ settings }) => settings.fontColor};
  cursor: pointer;
  text-transform: uppercase;
  text-decoration: underline;
`;

export const SelectWrapper = styled.section<{ settings: SettingsType }>`
  position: relative;
  width: ${({ settings }) => settings.width};
  overflow: hidden;
  margin: auto;
  background: ${({ settings }) => settings.backgroundColor};
  border-radius: 5px;
`;

export const SelectContent = styled.div<{ settings: SettingsType }>`
  display: flex;
  position: relative;
  justify-content: flex-start;
  flex-flow: row nowrap;
  flex-shrink: 1;
  flex-grow: 0;
  background: ${({ settings }) => settings.backgroundColor};
  overflow: hidden;
  left: calc(50% - ${ITEM_WIDTH / 2}px); // active item in the middle
`;

export const SelectOptions = styled.div<{ settings: SettingsType }>`
  position: relative;
  background: ${({ settings }) => settings.backgroundColor};
  cursor: pointer;
  height: ${({ settings }) => settings.height};
  line-height: ${({ settings }) => settings.height};
  flex-shrink: 0;
  width: ${ITEM_WIDTH}px;
`;

export const SelectedTag = styled.span<{ settings: SettingsType }>`
  position: absolute;
  width: 100%;
  top: 15px;
  display: block;
  line-height: 1;
  color: ${({ settings }) => settings.activeFontColor};
  font-size: calc(${({ settings }) => settings.fontSize} - 12px);
`;

export const SelectOption = styled.span<{
  isActive: boolean;
  settings: SettingsType;
  opacity: number;
}>`
  position: relative;
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  height: 100%;
  width: ${ITEM_WIDTH}px;
  text-align: center;
  font-size: ${({ isActive, settings }) =>
    isActive ? `calc(${settings.fontSize} + 4px)` : settings.fontSize};
  font-weight: 400;
  margin: 0;
  color: ${({ isActive, settings }) =>
    isActive ? settings.activeFontColor : settings.fontColor};
  transition: font-size 0.3s ease-out;
  opacity: ${({ opacity }) => opacity};
`;

export const SubmitButton = styled.button<{ settings: SettingsType }>`
  margin-top: 30px;
  padding: 10px 20px;
  border-radius: 5px;
  border: none;
  text-transform: uppercase;
  background: blue;
  color: white;
`;
