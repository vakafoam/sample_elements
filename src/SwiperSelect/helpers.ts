import { SETTINGS, SettingsType } from "SwiperSelect/constants";
import { OptionType } from "SwiperSelect";

export const getFinalSettings = (settings?: SettingsType): SettingsType =>
  settings ? { ...SETTINGS, ...settings } : SETTINGS;

export const getSelectedOption = (options: OptionType[]): OptionType => {
  const found = options.find((o: OptionType) => o.selected);
  return found ? found : options[0];
};

export const findSelectedItemIndex = (options: OptionType[]): number => {
  let selected = options.findIndex(el => {
    if (el.selected) return true;
    return false;
  });
  if (selected === -1) selected = 0;
  return selected;
};
