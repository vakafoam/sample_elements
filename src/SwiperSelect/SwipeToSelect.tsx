import React, { useState, useEffect, KeyboardEvent } from "react";
import styled from "../style/styled-components";
import { ITEM_WIDTH, SETTINGS } from "./constants";
import { findSelectedItemIndex } from "./helpers";
import { OptionType, SettingsType } from "./";

const SelectWrapper = styled.section`
  // display: flex;
  width: 80%;
  position: relative;
  overflow: hidden;
  margin: auto;
  background: black;
`;

const SelectContent = styled.div`
  display: flex;
  position: relative;
  justify-content: flex-start;
  flex-flow: row nowrap;
  flex-shrink: 1;
  flex-grow: 0;
  background: pink;
  overflow: hidden;
  left: calc(50% - ${ITEM_WIDTH / 2}px); // active item in the middle
`;

const SelectOptions = styled.div`
  display: flex;
  position: relative;
  background: white;
  //   align-items: center;
  cursor: pointer;
  justify-content: center;

  // From info element
  position: relative;
  //   display: block;
  flex-direction: column;
  background: yellow;
  height: 80px;
  flex-shrink: 0;
  width: ${ITEM_WIDTH}px;
`;

const TimelineInfo = styled.div`
  position: relative;
  display: block;
  background: yellow;
  height: 100%;
  flex-shrink: 0;
  width: ${ITEM_WIDTH}px;
`;

const SelectedTag = styled.span`
  position: absolute;
  width: 100%;
  top: 5px;
  display: block;
  justify-self: flex-start;
  text-align: center;
  font-size: 12px;
`;

const SelectOption = styled.p<{ isActive: boolean }>`
  box-sizing: border-box;
  justify-self: center;
  line-height: 1;
  text-align: center;
  position: relative;
  font-size: ${({ isActive }) => (isActive ? "26px" : "24px")};
  font-weight: 400;
  margin: 0;
  color: ${({ isActive }) => (isActive ? "blue" : "black")};
  transition: font-size 0.1s ease-out;
  transition-delay: 0.2s;
`;

interface PropsType {
  options: OptionType[];
  settings: SettingsType;
}

const SwipeToSelect = ({ options, settings }: PropsType) => {
  const [itemsNum, setItemsNum] = useState<number>(0);
  const [i, setI] = useState<number>(0);
  const [x0, setX0] = useState<number | null>(null);
  const [tx, setTx] = useState<number>(0);
  const [isDragging, setIsDragging] = useState<boolean>(false);
  const [selected, setSelected] = useState<number | null>(null);
  const [transformValue, setTransformValue] = useState<string>();

  useEffect(() => {
    const selectedIndex = findSelectedItemIndex(options);
    setI(selectedIndex);
    setSelected(selectedIndex);
    if (options.length) setItemsNum(options.length);
  }, []);

  useEffect(() => {
    const newTransform = `translate(${(i * -100) /
      itemsNum}%) translate(${tx}px)`;
    setTransformValue(newTransform);
  }, [i, tx]);

  // Lock the Options from moving to the next slide while user drags
  const lock = (e: any) => {
    const X0 = unify(e).clientX;
    setX0(X0);
    setIsDragging(true);
  };

  //   unlock = e => {
  //     const x0 = this.unify(e).clientX;
  //     this.setState({ x0, isDragging: false });
  //   };

  const isItemActive = (id: number) => {
    const el = document.getElementById(`item_${id}`);
    console.log("Element", el && el.getBoundingClientRect());
  };

  const move = (e: any, idx?: number) => {
    const dx = unify(e).clientX - (x0 || 0);
    if (idx !== undefined && Math.abs(dx) < 5) {
      // handle only click
      setI(idx);
      setIsDragging(false);
    } else if (isDragging) {
      const s = Math.round((dx * 1) / ITEM_WIDTH);

      if ((i > 0 || s < 0) && (i < itemsNum - 1 || s > 0)) {
        const newI = i - s < 0 ? 0 : i - s >= itemsNum ? itemsNum - 1 : i - s; // not sliding beyond min-max values
        setI(newI);
      }
      setTx(0);
      setIsDragging(false);
      setX0(null);
    }
  };

  // Get the right event object depending on if it's Touch or Mouse
  const unify = (e: any) => {
    return e.changedTouches ? e.changedTouches[0] : e;
  };

  // Manage the UI response while user drags the Timeline
  const drag = (e: any) => {
    if (isDragging) {
      // console.log("EVENT", e);
      if (!e.changedTouches) e.preventDefault();
      e.stopPropagation();

      const tX = unify(e).clientX - (x0 || 0);
      setTx(tX);
    }
  };

  const updateItem = (e: any, idx: number) => {
    e.stopPropagation();
    // TODO: check if dragging, then don't update
    const dx = unify(e).clientX - (x0 || 0);
    console.log("click: ", dx);

    if (idx && Math.abs(dx)) {
      // handle only click
      console.log("IDX", idx);
      setI(idx);
      setIsDragging(false);
    }
    //   this.setState({ i: index  })
  };

  const onKeyDown = (e: KeyboardEvent) => {
    if (e.keyCode === 37 || e.keyCode === 40) {
      if (i === 0) return;
      setI(i - 1);
    } else if (e.keyCode === 38 || e.keyCode === 39) {
      if (i === itemsNum - 1) return;
      setI(i + 1);
    }
  };

  // const onScroll = (e: any) => {
  //   console.log("Scroll", e);
  //   e = window.event || e;
  //   var delta = Math.max(-1, Math.min(1, e.wheelDelta || -e.detail));
  //   // @ts-ignore
  //   document.getElementById("selectContent").scrollLeft -= delta * 40; // Multiplied by 40
  //   e.preventDefault();
  // };

  const getContentStyles = () => {
    return {
      width: `${itemsNum * ITEM_WIDTH}px`,
      transform: transformValue,
      transition: isDragging ? "none" : "transform 0.5s ease-out",
    };
  };

  return (
    <SelectWrapper>
      <SelectContent
        // id="selectContent"
        onMouseDown={lock}
        //   onMouseUp={this.move}
        // onMouseUp={this.unlock}
        onMouseMove={drag}
        onTouchStart={lock}
        onTouchEnd={move}
        onTouchMove={drag}
        onKeyDown={onKeyDown}
        onMouseLeave={move}
        style={getContentStyles()}
        // onScroll={onScroll}
        tabIndex={0}
      >
        {options.map(({ value, text }, idx) => (
          <SelectOptions id={`item_${idx}`} key={`item_${idx}`}>
            {selected === i && i === idx && (
              <SelectedTag>Current choice</SelectedTag>
            )}
            <SelectOption
              onClick={e => move(e, idx)}
              onDoubleClick={e => {
                e.preventDefault();
                move(e, idx);
              }}
              isActive={i === idx}
              // isActive={this.isItemActive(idx)}
            >
              {text}
            </SelectOption>
          </SelectOptions>
        ))}
      </SelectContent>
    </SelectWrapper>
  );
};

export default SwipeToSelect;
