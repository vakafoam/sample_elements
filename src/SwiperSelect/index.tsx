import React, { useState, useEffect } from "react";
import styled from "../style/styled-components";
import SwipeToSelect from "./SwipeToSelect.jsx";

import { getFinalSettings, getSelectedOption } from "./helpers";

export interface SettingsType {
  width?: string;
  height?: string;
  backgroundColor?: string;
  fontColor?: string;
  fontFamily?: string;
  fontSize?: string;
  units?: string;
}

interface SettingsPropsType {
  settings: SettingsType;
}

const SelectedState = styled.div<{ settings: SettingsType }>`
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  width: ${({ settings }) => settings.width};
  height: ${({ settings }) => settings.height};
  background: ${({ settings }) => settings.backgroundColor};
  color: ${({ settings }) => settings.fontColor};
  font-size: ${({ settings }) => settings.fontSize};
  margin: auto;
  padding: 10px;
  border-radius: 5px;
`;

const SelectedValue = styled.span<{ settings: SettingsType }>`
  display: flex;
  flex-direction: column;
  text-align: left;
  justify-content: center;
  margin-left: ${({ settings }) => settings.fontSize};
`;
const ValueUnits = styled.span<{ settings: SettingsType }>`
  font-size: calc(${({ settings }) => settings.fontSize} - 8px);
`;
const EnableEdit = styled.button<{ settings: SettingsType }>`
  position: relative;
  display: block;
  justify-self: flex-end;
  align-self: center;
  border: none;
  padding: 20px;
  background-color: transparent;
  color: ${({ settings }) => settings.fontColor};
  cursor: pointer;
  text-transform: uppercase;
  text-decoration: underline;
`;

export interface OptionType {
  value: string | number | boolean;
  text?: string | number | boolean;
  selected?: boolean;
}

export interface PropsType {
  options: OptionType[];
  settings?: SettingsType;
}

const SwiperSelect = ({ options, settings }: PropsType) => {
  const [editMode, setEditMode] = useState<boolean>(false);
  const [selectedOption, setSelectedOption] = useState<OptionType>();
  const finalSettings = getFinalSettings(settings);

  useEffect(() => {
    const selected = getSelectedOption(options);
    setSelectedOption(selected);
  }, []);

  return editMode ? (
    <SwipeToSelect options={options} settings={finalSettings} />
  ) : (
    <SelectedState settings={finalSettings}>
      <SelectedValue settings={finalSettings}>
        {selectedOption && selectedOption.text}
        <ValueUnits settings={finalSettings}>{finalSettings.units}</ValueUnits>
      </SelectedValue>

      <EnableEdit onClick={() => setEditMode(true)} settings={finalSettings}>
        Edit
      </EnableEdit>
    </SelectedState>
  );
};

export default SwiperSelect;
