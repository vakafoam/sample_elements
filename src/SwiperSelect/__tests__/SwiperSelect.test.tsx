import React from "react";
import { shallow, mount } from "enzyme";
import SwiperSelect, { OptionType } from "SwiperSelect";
import SwipeToSelect from "SwiperSelect/SwipeToSelect";
import { EnableEdit, SubmitButton } from "SwiperSelect/componentsStyled";
import { SETTINGS } from "SwiperSelect/constants";

const options: OptionType[] = [
  { value: "2121", text: "2121" },
  { value: "2122", text: "2122", selected: false },
  { value: "2125", text: "2125", selected: true },
  { value: "2126", text: "2126", selected: true },
];

describe("Test EnableEdit button", () => {
  it("Test click event", () => {
    const mockCallBack = jest.fn();

    const button = mount(
      <EnableEdit settings={SETTINGS} onClick={mockCallBack} />,
    );

    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe("Test Submit button", () => {
  it("Test click event", () => {
    const mockCallBack = jest.fn();

    const button = mount(
      <SubmitButton type="submit" onClick={mockCallBack} settings={SETTINGS} />,
    );

    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe("SwiperSelect", () => {
  it("should render correctly and match the snapshot", () => {
    const component = shallow(
      <SwiperSelect options={options} onSubmit={() => {}} />,
    );

    expect(component).toMatchSnapshot();
  });
});

describe("SwipeToSelect", () => {
  it("should render correctly and match the snapshot", () => {
    const component = shallow(
      <SwipeToSelect
        options={options}
        settings={SETTINGS}
        onSubmit={() => {}}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
