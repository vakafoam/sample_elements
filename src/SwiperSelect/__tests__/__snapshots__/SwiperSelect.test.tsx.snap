// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeToSelect should render correctly and match the snapshot 1`] = `ShallowWrapper {}`;

exports[`SwiperSelect should render correctly and match the snapshot 1`] = `ShallowWrapper {}`;
